src/main/io/serial_4way.c

```diff
@@ -428,487 +411,402 @@ void esc4wayProcess(serialPort_t *mspPort)
-                                ACK_OUT = ACK_D_GENERAL_ERROR;
+            // ******* Interface related stuff *******
+            case cmd_InterfaceTestAlive: {
+                if (isMcuConnected()) {
+                    switch (CurrentInterfaceMode) {
+#ifdef USE_SERIAL_4WAY_BLHELI_BOOTLOADER
+                    case imATM_BLB:
+                    case imSIL_BLB:
+                    case imARM_BLB: {
+                        if (!BL_SendCMDKeepAlive()) { // SetStateDisconnected() included
+                            ACK_OUT = ACK_D_GENERAL_ERROR;
                         }
-                        if ( ACK_OUT != ACK_OK) SET_DISCONNECTED;
+                        break;
+                    }
+#endif
+#ifdef USE_SERIAL_4WAY_SK_BOOTLOADER
+                    case imSK: {
+                        if (!Stk_SignOn()) { // SetStateDisconnected();
+                            ACK_OUT = ACK_D_GENERAL_ERROR;
+                        }
+                        break;
+                    }
+#endif
+                    default:
+                        ACK_OUT = ACK_D_GENERAL_ERROR;
+                    }
+                    if ( ACK_OUT != ACK_OK) SET_DISCONNECTED;
+                }
+                break;
+            }
+            case cmd_ProtocolGetVersion: {
+                // Only interface itself, no matter what Device
+                Dummy.bytes[0] = SERIAL_4WAY_PROTOCOL_VER;
+                break;
+            }
+            case cmd_InterfaceGetName: {
+                // Only interface itself, no matter what Device
+                // O_PARAM_LEN=16;
+                O_PARAM_LEN = strlen(SERIAL_4WAY_INTERFACE_NAME_STR);
+                O_PARAM = (uint8_t *)SERIAL_4WAY_INTERFACE_NAME_STR;
+                break;
+            }
+            case cmd_InterfaceGetVersion: {
+                // Only interface itself, no matter what Device
+                // Dummy = iUart_res_InterfVersion;
+                O_PARAM_LEN = 2;
+                Dummy.bytes[0] = SERIAL_4WAY_VERSION_HI;
+                Dummy.bytes[1] = SERIAL_4WAY_VERSION_LO;
+                break;
+            }
+            case cmd_InterfaceExit: {
+                isExitScheduled = true;
+                break;
+            }
+            case cmd_InterfaceSetMode: {
+#if defined(USE_SERIAL_4WAY_BLHELI_BOOTLOADER) && defined(USE_SERIAL_4WAY_SK_BOOTLOADER)
+                if ((ParamBuf[0] <= imARM_BLB) && (ParamBuf[0] >= imSIL_BLB)) {
+#elif defined(USE_SERIAL_4WAY_BLHELI_BOOTLOADER)
+                if (((ParamBuf[0] <= imATM_BLB) || (ParamBuf[0] == imARM_BLB)) && (ParamBuf[0] >= imSIL_BLB)) {
+#elif defined(USE_SERIAL_4WAY_SK_BOOTLOADER)
+                if (ParamBuf[0] == imSK) {
+#endif
+                    CurrentInterfaceMode = ParamBuf[0];
+                } else {
+                    ACK_OUT = ACK_I_INVALID_PARAM;
+                }
+                break;
+            }
+            case cmd_DeviceReset: {
+                bool rebootEsc = false;
```

Comment:
Codacy found an issue: [The scope of the variable 'rebootEsc' can be reduced.](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6307959)